UseCases/Sources/HTTPEndToEnd/InstrumentedHTTPClient.swift

```diff
@@ -6,21 +6,36 @@ import NIO
 import NIOHTTP1
 
 struct InstrumentedHTTPClient {
-    private let client = HTTPClient(eventLoopGroupProvider: .createNew)
+    private let client: HTTPClient
     private let instrument: Instrument<HTTPHeaders, HTTPHeaders>
 
     init<I>(instrument: I, eventLoopGroupProvider: HTTPClient.EventLoopGroupProvider)
         where
         I: InstrumentProtocol,
         I.InjectInto == HTTPHeaders,
         I.ExtractFrom == HTTPHeaders {
+        self.client = HTTPClient(eventLoopGroupProvider: eventLoopGroupProvider)
         self.instrument = Instrument(instrument)
     }
 
+    // TODO: deadline: NIODeadline? would move into baggage?
+    public func get(url: String, baggage: BaggageContext = .init()) -> EventLoopFuture<HTTPClient.Response> {
+        do {
+            let request = try HTTPClient.Request(url: url, method: .GET)
+            return self.execute(request: request, baggage: baggage)
+        } catch {
+            return self.client.eventLoopGroup.next().makeFailedFuture(error)
+        }
+    }
+
     func execute(request: HTTPClient.Request, baggage: BaggageContext) -> EventLoopFuture<HTTPClient.Response> {
         var request = request
         self.instrument.inject(from: baggage, into: &request.headers)
         baggage.logger.info("AsyncHTTPClient: Execute request")
         return self.client.execute(request: request)
     }
+
+    func syncShutdown() throws {
+        try self.client.syncShutdown()
+    }
```

Comment:
this cannot be called on an event loop, which makes it hard it the Handler is the "only" place that is able to call shutdown -- we could not call it in there after all. Thus I propose to pass in the client to handlers